crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1499,15 +1499,48 @@ async fn filter_payment_mandate_based(
     Ok(recurring_filter)
 }
 
+pub async fn do_list_customer_pm_fetch_customer_if_not_passed(
+    state: &routes::AppState,
+    merchant_account: domain::MerchantAccount,
+    key_store: domain::MerchantKeyStore,
+    req: Option<api::PaymentMethodListRequest>,
+    customer_id: Option<&str>,
+) -> errors::RouterResponse<api::CustomerPaymentMethodsListResponse> {
+    let db = &*state.store;
+    if customer_id.is_some() {
+        let customer_id = customer_id.ok_or(errors::ApiErrorResponse::CustomerNotFound)?;
+        list_customer_payment_method(state, merchant_account, key_store, None, customer_id).await
+    } else {
+        let cloned_secret = req.and_then(|r| r.client_secret.as_ref().cloned());
```

Comment:
We are using only `client_secret` from the request, what about other fields in `PaymentMethodsListRequest` ? Don't we need them? If not we can remove them from the struct
https://github.com/juspay/hyperswitch/blob/6447b04574e941b9214239bf5b65b7c1a229dfd6/crates/api_models/src/payment_methods.rs#L324
cc: @bernard-eugine

Reply:
We need not remove the struct, but the fields can be removed.